java/autoscaler/src/main/java/com/example/bigtable/autoscaler/Autoscaler.java

```diff
@@ -0,0 +1,207 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.example.bigtable.autoscaler;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.text.SimpleDateFormat;
+import java.util.TimeZone;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import org.joda.time.DateTime;
+
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.monitoring.v3.Monitoring;
+import com.google.api.services.monitoring.v3.MonitoringScopes;
+import com.google.api.services.monitoring.v3.model.Point;
+import com.google.bigtable.admin.v2.Cluster;
+import com.google.cloud.bigtable.grpc.BigtableClusterName;
+import com.google.cloud.bigtable.grpc.BigtableClusterUtilities;
+
+/**
+ * An example that leverages <a href="https://cloud.google.com/bigtable/docs/">Cloud Bigtable</a>
+ * metrics from the <a href="https://cloud.google.com/monitoring/api/v3/">Stackdriver Monitoring
+ * API</a> in order to resize a Cloud Bigtable cluster. This example uses the server's CPU load
+ * which is an indicator of overall usage of a cluster. If a cluster is overloaded, performance
+ * drops. This example looks at the CPU load metric, which is the key indicator that a cluster is
+ * overloaded.  This example is not a comprehensive solution, but rather a starting point by which
+ * a product grade auto-scaler can be built that is targted to your specific situation.
+ */
+public class Autoscaler {
+
+  /**
+   * This metric represents the average CPU load of a cluster. More metrics can be found
+   * <a href="https://cloud.google.com/monitoring/api/metrics#gcp-bigtable">here</a>. Other metrics
+   * such as "server/latencies" and "server/sent_bytes_count" can also be useful
+   */
+  public static final String CPU_METRIC = "bigtable.googleapis.com/cluster/cpu_load";
+
+  /**
+   * The minimum number of nodes to use. The default minimum is 3. If you have a lot of data, the
+   * rule of thumb is to not go below 1 node per terabyte for SSD clusters, and 5 terabytes for HDD.
```

Comment:
FYI, Misha and Eric told me that the rule of thumb was 1 node per 2.5 TB for SSD and 1 node per 8 TB for HDD. It'd be good to confirm the numbers with them so this comment is consistent with our docs.